### PR TITLE
Fix tooling to compile pyopenms as a single module

### PR DESCRIPTION
**Pull request by poshul**
## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/pyOpenMS/CMakeLists.txt [hunk 1]
```diff
@@ -678,6 +678,10 @@ else()
         LIBRARY_OUTPUT_DIRECTORY_DEBUG "${CMAKE_BINARY_DIR}/pyOpenMS/pyopenms"
         RUNTIME_OUTPUT_DIRECTORY_DEBUG "${CMAKE_BINARY_DIR}/pyOpenMS/pyopenms"
     )
+    add_custom_command(TARGET pyopenms_compile
```
Comment (jpfeuffer) [inline src/pyOpenMS/CMakeLists.txt:681]: The idea is, that this is not needed since everything is in one module.

Comment (jpfeuffer) [reply src/pyOpenMS/CMakeLists.txt:681]: I cannot judge since I never tested but in theory python.typing should find all classes.

Comment (poshul) [reply src/pyOpenMS/CMakeLists.txt:681]: I'm out of my depth in how the resolution is supposed to work. But adding that seems to have solved the compilation problem, and changing the create_cpp_extension seems to fix the naming issue.

Comment (jpfeuffer) [reply src/pyOpenMS/CMakeLists.txt:681]: That is super strange. The pyi files should not have anything to do with compilation.
I guess it won't hurt.
Can you add a Todo: check if really necessary.?

Comment (poshul) [reply src/pyOpenMS/CMakeLists.txt:681]: Done.

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**